{
  error: TypeError: logger.error is not a function
      at C:\Users\Juan\Desktop\Microservicios-Arquitectura\auth-service\src\index.js:106:10
      at process.processTicksAndRejections (node:internal/process/task_queues:105:5),
  level: 'error',
  message: 'unhandledRejection: logger.error is not a function\n' +
    'TypeError: logger.error is not a function\n' +
    '    at C:\\Users\\Juan\\Desktop\\Microservicios-Arquitectura\\auth-service\\src\\index.js:106:10\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)',
  stack: 'TypeError: logger.error is not a function\n' +
    '    at C:\\Users\\Juan\\Desktop\\Microservicios-Arquitectura\\auth-service\\src\\index.js:106:10\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:105:5)',
  rejection: true,
  date: 'Tue Aug 26 2025 12:56:18 GMT+0200 (hora de verano de Europa central)',
  process: {
    pid: 18792,
    uid: null,
    gid: null,
    cwd: 'C:\\Users\\Juan\\Desktop\\Microservicios-Arquitectura\\auth-service',
    execPath: 'C:\\Program Files\\nodejs\\node.exe',
    version: 'v22.18.0',
    argv: [
      'C:\\Program Files\\nodejs\\node.exe',
      'C:\\Users\\Juan\\Desktop\\Microservicios-Arquitectura\\auth-service\\src\\index.js'
    ],
    memoryUsage: {
      rss: 90877952,
      heapTotal: 61288448,
      heapUsed: 24427008,
      external: 20598186,
      arrayBuffers: 18264951
    }
  },
  os: { loadavg: [ 0, 0, 0 ], uptime: 14666.921 },
  trace: [
    {
      column: 10,
      file: 'C:\\Users\\Juan\\Desktop\\Microservicios-Arquitectura\\auth-service\\src\\index.js',
      function: null,
      line: 106,
      method: null,
      native: false
    },
    {
      column: 5,
      file: 'node:internal/process/task_queues',
      function: 'process.processTicksAndRejections',
      line: 105,
      method: 'processTicksAndRejections',
      native: false
    }
  ],
  service: 'auth-service',
  environment: 'development',
  version: '1.0.0',
  timestamp: '2025-08-26 12:56:18:5618'
}
{
  service: 'auth-service',
  environment: 'development',
  version: '1.0.0',
  errorLabelSet: Set(0) {},
  errorResponse: {
    ok: 0,
    errmsg: 'command find requires authentication',
    code: 13,
    codeName: 'Unauthorized'
  },
  ok: 0,
  code: 13,
  codeName: 'Unauthorized',
  level: 'error',
  message: 'Error en registro: command find requires authentication',
  stack: 'MongoServerError: command find requires authentication\n' +
    '    at Connection.sendCommand (/usr/src/app/node_modules/mongodb/lib/cmap/connection.js:305:27)\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
    '    at async Connection.command (/usr/src/app/node_modules/mongodb/lib/cmap/connection.js:333:26)\n' +
    '    at async Server.command (/usr/src/app/node_modules/mongodb/lib/sdam/server.js:171:29)\n' +
    '    at async FindOperation.execute (/usr/src/app/node_modules/mongodb/lib/operations/find.js:36:16)\n' +
    '    at async tryOperation (/usr/src/app/node_modules/mongodb/lib/operations/execute_operation.js:207:20)\n' +
    '    at async executeOperation (/usr/src/app/node_modules/mongodb/lib/operations/execute_operation.js:75:16)\n' +
    '    at async FindCursor._initialize (/usr/src/app/node_modules/mongodb/lib/cursor/find_cursor.js:61:26)\n' +
    '    at async FindCursor.cursorInit (/usr/src/app/node_modules/mongodb/lib/cursor/abstract_cursor.js:632:27)\n' +
    '    at async FindCursor.fetchBatch (/usr/src/app/node_modules/mongodb/lib/cursor/abstract_cursor.js:666:13)',
  timestamp: '2025-08-26 11:01:37:137'
}
{
  service: 'auth-service',
  environment: 'development',
  version: '1.0.0',
  errorLabelSet: Set(0) {},
  errorResponse: {
    ok: 0,
    errmsg: 'command find requires authentication',
    code: 13,
    codeName: 'Unauthorized'
  },
  ok: 0,
  code: 13,
  codeName: 'Unauthorized',
  level: 'error',
  message: 'Error en registro: command find requires authentication',
  stack: 'MongoServerError: command find requires authentication\n' +
    '    at Connection.sendCommand (/usr/src/app/node_modules/mongodb/lib/cmap/connection.js:305:27)\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n' +
    '    at async Connection.command (/usr/src/app/node_modules/mongodb/lib/cmap/connection.js:333:26)\n' +
    '    at async Server.command (/usr/src/app/node_modules/mongodb/lib/sdam/server.js:171:29)\n' +
    '    at async FindOperation.execute (/usr/src/app/node_modules/mongodb/lib/operations/find.js:36:16)\n' +
    '    at async tryOperation (/usr/src/app/node_modules/mongodb/lib/operations/execute_operation.js:207:20)\n' +
    '    at async executeOperation (/usr/src/app/node_modules/mongodb/lib/operations/execute_operation.js:75:16)\n' +
    '    at async FindCursor._initialize (/usr/src/app/node_modules/mongodb/lib/cursor/find_cursor.js:61:26)\n' +
    '    at async FindCursor.cursorInit (/usr/src/app/node_modules/mongodb/lib/cursor/abstract_cursor.js:632:27)\n' +
    '    at async FindCursor.fetchBatch (/usr/src/app/node_modules/mongodb/lib/cursor/abstract_cursor.js:666:13)',
  timestamp: '2025-08-26 11:02:50:250'
}
{
  service: 'auth-service',
  environment: 'development',
  version: '1.0.0',
  message: 'Error capturado: Error interno del servidor',
  stack: 'Error: Error interno del servidor\n' +
    '    at createError (/usr/src/app/src/utils/errors.js:12:10)\n' +
    '    at /usr/src/app/src/routes/auth.js:116:10\n' +
    '    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)',
  url: '/api/auth/register',
  method: 'POST',
  ip: '::ffff:172.20.0.1',
  userAgent: 'curl/8.15.0',
  userId: undefined,
  timestamp: '2025-08-26 11:02:50:250',
  level: 'error'
}
{
  service: 'auth-service',
  environment: 'development',
  version: '1.0.0',
  message: 'Error capturado: request aborted',
  stack: 'BadRequestError: request aborted\n' +
    '    at IncomingMessage.onAborted (/usr/src/app/node_modules/raw-body/index.js:245:10)\n' +
    '    at IncomingMessage.emit (node:events:517:28)\n' +
    '    at IncomingMessage._destroy (node:_http_incoming:224:10)\n' +
    '    at _destroy (node:internal/streams/destroy:109:10)\n' +
    '    at IncomingMessage.destroy (node:internal/streams/destroy:71:5)\n' +
    '    at abortIncoming (node:_http_server:781:9)\n' +
    '    at socketOnClose (node:_http_server:775:3)\n' +
    '    at Socket.emit (node:events:529:35)\n' +
    '    at TCP.<anonymous> (node:net:350:12)',
  url: '/register',
  method: 'POST',
  ip: '::ffff:172.20.0.4',
  userAgent: 'curl/8.15.0',
  userId: undefined,
  timestamp: '2025-08-26 11:04:22:422',
  level: 'error'
}
{
  service: 'auth-service',
  environment: 'development',
  version: '1.0.0',
  message: 'Error capturado: request aborted',
  stack: 'BadRequestError: request aborted\n' +
    '    at IncomingMessage.onAborted (/usr/src/app/node_modules/raw-body/index.js:245:10)\n' +
    '    at IncomingMessage.emit (node:events:517:28)\n' +
    '    at IncomingMessage._destroy (node:_http_incoming:224:10)\n' +
    '    at _destroy (node:internal/streams/destroy:109:10)\n' +
    '    at IncomingMessage.destroy (node:internal/streams/destroy:71:5)\n' +
    '    at abortIncoming (node:_http_server:781:9)\n' +
    '    at socketOnClose (node:_http_server:775:3)\n' +
    '    at Socket.emit (node:events:529:35)\n' +
    '    at TCP.<anonymous> (node:net:350:12)',
  url: '/api/auth/register',
  method: 'POST',
  ip: '::ffff:172.20.0.4',
  userAgent: 'curl/8.15.0',
  userId: undefined,
  timestamp: '2025-08-26 11:11:40:1140',
  level: 'error'
}
{
  service: 'auth-service',
  environment: 'development',
  version: '1.0.0',
  message: 'Error capturado: request aborted',
  stack: 'BadRequestError: request aborted\n' +
    '    at IncomingMessage.onAborted (/usr/src/app/node_modules/raw-body/index.js:245:10)\n' +
    '    at IncomingMessage.emit (node:events:517:28)\n' +
    '    at IncomingMessage._destroy (node:_http_incoming:224:10)\n' +
    '    at _destroy (node:internal/streams/destroy:109:10)\n' +
    '    at IncomingMessage.destroy (node:internal/streams/destroy:71:5)\n' +
    '    at abortIncoming (node:_http_server:781:9)\n' +
    '    at socketOnClose (node:_http_server:775:3)\n' +
    '    at Socket.emit (node:events:529:35)\n' +
    '    at TCP.<anonymous> (node:net:350:12)',
  url: '/api/auth/login',
  method: 'POST',
  ip: '::ffff:172.20.0.4',
  userAgent: 'curl/8.15.0',
  userId: undefined,
  timestamp: '2025-08-26 11:12:10:1210',
  level: 'error'
}
{
  service: 'auth-service',
  environment: 'development',
  version: '1.0.0',
  message: 'Error capturado: request aborted',
  stack: 'BadRequestError: request aborted\n' +
    '    at IncomingMessage.onAborted (/usr/src/app/node_modules/raw-body/index.js:245:10)\n' +
    '    at IncomingMessage.emit (node:events:517:28)\n' +
    '    at IncomingMessage._destroy (node:_http_incoming:224:10)\n' +
    '    at _destroy (node:internal/streams/destroy:109:10)\n' +
    '    at IncomingMessage.destroy (node:internal/streams/destroy:71:5)\n' +
    '    at abortIncoming (node:_http_server:781:9)\n' +
    '    at socketOnClose (node:_http_server:775:3)\n' +
    '    at Socket.emit (node:events:529:35)\n' +
    '    at TCP.<anonymous> (node:net:350:12)',
  url: '/api/auth/login',
  method: 'POST',
  ip: '::ffff:172.20.0.4',
  userAgent: 'curl/8.15.0',
  userId: undefined,
  timestamp: '2025-08-26 11:12:47:1247',
  level: 'error'
}
{
  service: 'auth-service',
  environment: 'development',
  version: '1.0.0',
  message: 'Error capturado: request aborted',
  stack: 'BadRequestError: request aborted\n' +
    '    at IncomingMessage.onAborted (/usr/src/app/node_modules/raw-body/index.js:245:10)\n' +
    '    at IncomingMessage.emit (node:events:517:28)\n' +
    '    at IncomingMessage._destroy (node:_http_incoming:224:10)\n' +
    '    at _destroy (node:internal/streams/destroy:109:10)\n' +
    '    at IncomingMessage.destroy (node:internal/streams/destroy:71:5)\n' +
    '    at abortIncoming (node:_http_server:781:9)\n' +
    '    at socketOnClose (node:_http_server:775:3)\n' +
    '    at Socket.emit (node:events:529:35)\n' +
    '    at TCP.<anonymous> (node:net:350:12)',
  url: '/api/auth/login',
  method: 'POST',
  ip: '::ffff:172.20.0.4',
  userAgent: 'curl/8.15.0',
  userId: undefined,
  timestamp: '2025-08-26 11:14:10:1410',
  level: 'error'
}
{
  service: 'auth-service',
  environment: 'development',
  version: '1.0.0',
  message: 'Error capturado: request aborted',
  stack: 'BadRequestError: request aborted\n' +
    '    at IncomingMessage.onAborted (/usr/src/app/node_modules/raw-body/index.js:245:10)\n' +
    '    at IncomingMessage.emit (node:events:517:28)\n' +
    '    at IncomingMessage._destroy (node:_http_incoming:224:10)\n' +
    '    at _destroy (node:internal/streams/destroy:109:10)\n' +
    '    at IncomingMessage.destroy (node:internal/streams/destroy:71:5)\n' +
    '    at abortIncoming (node:_http_server:781:9)\n' +
    '    at socketOnClose (node:_http_server:775:3)\n' +
    '    at Socket.emit (node:events:529:35)\n' +
    '    at TCP.<anonymous> (node:net:350:12)',
  url: '/api/auth/health',
  method: 'GET',
  ip: '::ffff:172.20.0.4',
  userAgent: 'PostmanRuntime/7.45.0',
  userId: undefined,
  timestamp: '2025-08-26 11:16:53:1653',
  level: 'error'
}
